# US Census Bureau COVID-19 Site  
_American Community Survey_
___

__June 2020 Income and Benfefits by U.S. Counties__

This dataset shows household income ranges and cutoffs. This is shown by county boundaries. This service is updated annually to contain the most currently released American Community Survey (ACS) 5-year data, and contains estimates and margins of error. The data is from 12 June 2020. 

This dataset was taken from the COVID-19 Census Site resources to collect accurate and updated Census information during the pandemic. 

__Sources__

In [ ]:
# source: https://covid19.census.gov/datasets/56f051341b4a40aa9aa5e2c33f85547a_2/data?geometry=167.484%2C-16.868%2C-166.852%2C72.108

In [ ]:
# Import Census data and Geographies
# List of Geographies and codes: https://jtleider.github.io/censusdata/geographies.html

from census import Census
from us import states
c = Census("cd1f52ad7e8ea45d92f2f657f56da781e3687cc8b")

# importing additional libraries
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
sns.set()
import sklearn

#  importing the censusdata and pandas modules, and setting some display options in pandas

import pandas as pd
import censusdata
pd.set_option('display.expand_frame_repr', False)
pd.set_option('display.precision', 2)

<b> Import Modules </b>

In [ ]:
# Imported packages
import numpy as np
import statistics
import pandas as pd
import matplotlib.pyplot as plt 
%matplotlib inline
import seaborn as sns
from numpy import array
from numpy import cov
import warnings
warnings.filterwarnings('ignore')

<b> Left align all markdown tables </b>

In [ ]:
%%html
<style>
    table {
        display: inline-block
    }
</style>

<b> Read csv file to a pandas dataframe </b>

In [ ]:
# load in data
df_count = pd.read_csv("Capstone-Project/economic_county_data.csv")

df_count.head()

<b> Number of rows, number of columns in dataset </b>

In [36]:
print('There are # number of rows in the dataset    :', df_count.shape[0])
print('There are # number of columns in the dataset :', df_count.shape[1])

There are # number of rows in the dataset    : 3233
There are # number of columns in the dataset : 28


| **# Rows** | **# Features** |
|----------|--------------|
| 12330    | 18           |


<b> Features datatypes and non-null </b>

In [17]:
#df_count.info()
# convert table to markdown, source: https://tableconvert.com/

| Variable                 | # values            |  Data Type |
|--------------------------|--------------------|---------|
| OBJECTID           | 3233 non\-null   | int64    |
| GEO\_NAME          |  3233 non\-null  | object  |
| GEO\_PARENT\_NAME  |   3233 non\-null | object |
| B11001\_001E       |  3220 non\-null  | float64 |
| B11001\_001M       |  3220 non\-null  | float64 |
| DP03\_0052E        |  3219 non\-null  | float64 |
| DP03\_0052M        |  3219 non\-null  | float64 |
| DP03\_0053E        |  3219 non\-null  | float64 |
| DP03\_0053M        |  3219 non\-null  | float64 |
| DP03\_0054E        |  3219 non\-null  | float64 |
| DP03\_0054M        |  3219 non\-null  | float64 |
| DP03\_0055E        |  3219 non\-null  | float64 |
| DP03\_0055M        |  3219 non\-null  | float64 |
| DP03\_0056E        |  3219 non\-null  | float64 |
| DP03\_0056M        |  3219 non\-null  | float64 |
| DP03\_0057E        |  3219 non\-null  | float64 |
| DP03\_0057M        |  3219 non\-null  | float64 |
| DP03\_0058E        |  3219 non\-null  | float64 |
| DP03\_0058M        |  3219 non\-null  | float64 |
| DP03\_0070E        |  3219 non\-null  | float64 |
| DP03\_0070M        |  3219 non\-null  | float64 |
| DP03\_0072E        |  3219 non\-null  | float64 |
| DP03\_0072M        |  3219 non\-null  | float64 |
| DP03\_0074E        |  3219 non\-null  | float64 |
| DP03\_0074M        |  3219 non\-null  | float64 |
| HOUSELT75KP\_CALC  |  3219 non\-null  | float64 |
| INCLT75E\_CALC     |  3219 non\-null  | float64 |
| INCLT75M\_CALC     |  3219 non\-null  | float64 |

## Take-Away
* All features and the target variable have no null values

<b> Feature Desciptions </b>

In [ ]:
# Variable List Desciptions 
    #B11001_001E Total Households
    #B11001_001M Total Households margin of error 
    #DP03_0052M Total Households with income - Less then $10,000
    #DP03_0052M Total Households with income - Less then $10,000 margin of error 
    #DP03_0053E Total Households with income - $10,000 to $14,999
    #DP03_0053M Total Households with income - $10,000 to $14,999 margin of error
    #DP03_0054E Total Households with income - $15,000 to $24,999
    #DP03_0054M Total Households with income - $15,000 to $24,999 margin of error
    #DP03_0055E Total Households with income - $25,000 to $34,999
    #DP03_0055M Total Households with income - $25,000 to $34,999 margin of error
    #DP03_0056E Total Households with income - $35,000 to $49,999
    #DP03_0056M Total Households with income - $35,000 to $49,999 margin of error
    #DP03_0057E Total Households with income - $50,000 to $74,999
    #DP03_0057M Total Households with income - $50,000 to $74,999 margin of error
    #DP03_0058E Total Households with income - $75,000 to $99,999
    #DP03_0058M Total Households with income - $75,000 to $99,999 margin of error
    #DP03_0070E Total Households - With Supplemental Security Income
    #DP03_0070M Total Households - With Supplemental Security Income margin of error 
    #DP03_0072E Total Households - With cash public assistance income
    #DP03_0072M Total Households - With cash public assistance income margin of error
    #DP03_0074E Total Households - With Food Stamp/SNAP benefits in the past 12 months
    #DP03_0074M Total Households - With Food Stamp/SNAP benefits in the past 12 months margin of error
    #HOUSELT75KP_CALC Percent of Household whose income in the past 12 months was less than $75,000
    #INCLT75E_CALC Total Households with income - less than $75,000
    #INCLT75M_CALC Total Households with income - less than $75,000 margin of error

<b> Descriptive Statistics </b>

In [19]:
df_count.describe().round(2)

OBJECTID  B11001_001E  B11001_001M  DP03_0052E  DP03_0052M  DP03_0053E  DP03_0053M  DP03_0054E  DP03_0054M  DP03_0055E  ...  DP03_0058M  DP03_0070E  DP03_0070M  DP03_0072E  DP03_0072M  DP03_0074E  DP03_0074M  HOUSELT75KP_CALC  INCLT75E_CALC  INCLT75M_CALC
count   3233.00     3.22e+03      3220.00      3219.0     3219.00     3219.00     3219.00     3219.00     3219.00     3219.00  ...     3219.00     3219.00     3219.00     3219.00     3219.00     3219.00     3219.00           3219.00       3.22e+03        3219.00
mean    1617.00     3.76e+04       387.36      2458.7      235.82     1755.09      204.71     3530.50      281.57     3498.90  ...      293.23     2005.20      199.49      944.46      133.06     4692.76      304.16             68.87       2.25e+04        1667.27
std      933.43     1.14e+05       384.07      7445.6      239.27     5235.47      200.98    10120.61      276.69     9970.79  ...      325.63     6685.05      209.23     3463.96      156.55    14536.99      310.99             10.75       6.49e+04        1659.80
min        1.00     3.30e+01         9.00         0.0        2.00        0.00        4.00        0.00        3.00        2.00  ...        3.00        0.00        3.00        0.00        2.00        0.00        3.00             24.20       1.50e+01          29.00
25%      809.00     4.32e+03       177.00       307.0       91.50      264.00       83.00      531.50      118.00      490.00  ...      107.00      224.50       73.50       84.00       45.00      550.00      120.00             63.22       3.07e+03         698.00
50%     1617.00     9.95e+03       283.00       754.0      171.00      622.00      151.00     1213.00      207.00     1136.00  ...      192.00      612.00      145.00      223.00       87.00     1430.00      223.00             69.97       7.20e+03        1202.00
75%     2425.00     2.54e+04       456.25      1931.0      296.00     1423.50      253.00     2812.50      342.00     2733.50  ...      344.00     1506.00      249.00      610.00      158.00     3570.00      378.00             75.79       1.72e+04        1999.50
max     3233.00     3.31e+06      5533.00    192319.0     2727.00   168753.00     2965.00   296864.00     3160.00   282438.00  ...     4013.00   226027.00     2811.00   117110.00     2153.00   296809.00     3498.00            100.00       1.86e+06       19774.00

[8 rows x 26 columns]

# Calculations
___

_Total Number of Households by State_

In [37]:
# group state counties together for total households 
# example: find out the total number of households in Albama 
total_house = df_count['B11001_001E'].groupby(df_count['GEO_PARENT_NAME'])
print(total_house.sum())

GEO_PARENT_NAME
Alabama                                         1.86e+06
Alaska                                          2.53e+05
American Samoa                                  0.00e+00
Arizona                                         2.52e+06
Arkansas                                        1.15e+06
California                                      1.30e+07
Colorado                                        2.11e+06
Commonwealth of the Northern Mariana Islands    0.00e+00
Connecticut                                     1.37e+06
Delaware                                        3.58e+05
District of Columbia                            2.81e+05
Florida                                         7.62e+06
Georgia                                         3.71e+06
Guam                                            0.00e+00
Hawaii                                          4.57e+05
Idaho                                           6.18e+05
Illinois                                        4.83e+06
Indiana        

_Total Number of Counties by State_

In [39]:
# total number of counties per state
total_county = df_count['GEO_PARENT_NAME'].value_counts()
print(total_county)

Texas                                           254
Georgia                                         159
Virginia                                        133
Kentucky                                        120
Missouri                                        115
Kansas                                          105
Illinois                                        102
North Carolina                                  100
Iowa                                             99
Tennessee                                        95
Nebraska                                         93
Indiana                                          92
Ohio                                             88
Minnesota                                        87
Michigan                                         83
Mississippi                                      82
Puerto Rico                                      78
Oklahoma                                         77
Arkansas                                         75
Wisconsin   

In [43]:
#total_county.plot(kind='barh')
# output too small to see all states

In [ ]:
ax = sns.boxplot(y="",x="GEO_PARENT_NAME", data=df_count)